libs/mongodb-client/Cargo.toml

```diff
@@ -6,4 +6,7 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-mongodb = "2.1.0"
\ No newline at end of file
+mongodb = "2.1.0"
+percent-encoding = "2.0.0"
+thiserror = "1"
+once_cell = "1"
```

Comment:
Nitpick, but maybe we could separate these deps to make it clear they can be removed once the io-free url parsing is available upstream

Reply:
Yep.